app/scripts/controllers/permissions/background-api.js

```diff
@@ -115,14 +112,7 @@ export function getPermissionBackgroundApiMethods({
           },
           permissions: {
             [RestrictedMethods.eth_accounts]: {},
-            [PermissionNames.permittedChains]: {
-              caveats: [
-                {
-                  type: CaveatTypes.restrictNetworkSwitching,
-                  value: [chainId],
-                },
-              ],
-            },
+            [PermissionNames.permittedChains]: {},
```

Comment:
I don' think I understand what is broken here. If I am correct, you start without Ethereum mainnet permissioned, and then switch to it. After approving the switch, Ethereum mainnet is checked inside the EditNetworkModal. I assume this dapp has EIP-1193 granted permissions, otherwise the switch chain call would have failed (rather than prompting the user to approve or not)

Reply:
Oh sorry I'm specifically highlighting that that switch chain permission UI no longer shows the ethereum avatar or network name

Reply:
Fixed here https://github.com/MetaMask/metamask-extension/pull/27471/commits/60f7518e5864a249c115deba8fcc8d3fabd45a34

will pull into PR against dev as well

Reply:
https://github.com/MetaMask/metamask-extension/pull/27518